test/tests/functional/pbs_multi_sched.py

```diff
@@ -2195,3 +2215,393 @@ def test_multi_sched_priority_sockets(self):
+        try:
+            rid = self.server.submit(r)
+        except PbsSubmitError as e:
+            msg = "resv attribute 'partition' is readonly"
+            self.server.log_match(msg)
+            self.assertTrue("hook 'h1' encountered an exception" in e.msg[0])
+
+    def test_resv_alter(self):
+        """
+        Test if a reservation confirmed by a multi-sched can be altered by the
+        same scheduler.
+        """
+        self.common_setup2()
+        # Submit 4 reservations to fill up the system and check they are
+        # confirmed
+        for _ in range(4):
+            a = {'Resource_List.select': '1:ncpus=2', 'reserve_start': int(
+                time.time() + 60), 'reserve_end': int(time.time() + 120)}
+            r = Reservation(TEST_USER, a)
+            rid = self.server.submit(r)
+            attr = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+            self.server.expect(RESV, attr, rid)
+            partition = self.server.status(RESV, 'partition', id=rid)
+            if (partition[0]['partition'] == 'P4'):
+                    old_end_time = a['reserve_end']
+                    modify_resv = rid
+        # Modify the endtime of reservation confirmed on partition P4 and
+        # make sure the node solution is correct.
+        end_time = old_end_time + 60
+        bu = BatchUtils()
+        new_end_time = bu.convert_seconds_to_datetime(end_time)
+        attrs = {'reserve_end': new_end_time}
+        time_now = time.time()
+        self.server.alterresv(modify_resv, attrs)
+        attr = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, attr, modify_resv)
+        partition = self.server.status(RESV, 'partition', id=modify_resv)
+        self.assertEqual(partition[0]['partition'], 'P4')
+        rnodes = {'resv_nodes': '(vnode[3]:ncpus=2)'}
+        self.server.expect(RESV, rnodes, id=modify_resv)
+        msg = modify_resv + ";Reservation Confirmed"
+        self.scheds['sc1'].log_match(msg, starttime=time_now)
+
+    def test_setting_default_partition(self):
+        """
+        Test if setting default partition on pbs scheduler/queue/node fails
+        """
+
+        self.common_setup()
+        a = {'partition': 'pbs-default'}
+        try:
+            self.server.manager(MGR_CMD_SET, QUEUE, a, id='workq')
+        except PbsManagerError as e:
+            self.assertIn("Default partition name is not allowed", e.msg[0])
+        try:
+            self.server.manager(MGR_CMD_SET, NODE, a, id='vnode[4]')
+        except PbsManagerError as e:
+            self.assertIn("Default partition name is not allowed", e.msg[0])
+        a = {'partition': 'new-part, pbs-default'}
+        try:
+            self.server.manager(MGR_CMD_SET, SCHED, a, id='sc1')
+        except PbsManagerError as e:
+            self.assertIn("Default partition name is not allowed", e.msg[0])
+
+    def degraded_resv_reconfirm(self, start, end, rrule=None, run=False):
```

Comment:
Instead of copying this function in from pbs_reservations.py, maybe move the tests that need it into pbs_reservations.py?

Reply:
This isn't a 1:1 copy of the function, I had to change it to add two nodes to the same partition and then check for resv_nodes accordingly. If you don't feel strongly about it, I'd like to retain it here.

Reply:
The original function handles cases for multiple tests.  Is it possible to add your changes to the original function?  I'd hate to have to have such code duplication just for a small change.